architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,392 @@
+
+**Developers** – Developers need to use the resource types available to them to model their applications and deploy to environments which they have access to.
+
+**Environment administrators** – An environment administrator needs to create and maintain environments in Radius for developers to deploy their applications to. The environment administrator must ensure the physical or cloud environment is prepared to have applications deployed to it and that the Radius environment is configured correctly. In some organizations, the environment administrator will determine who has permission to deploy applications to which environment.
+
+**Resource type administrators** – A resource type administrator needs to maintain the resource types which are available in the Radius system and manage who can use those resource types within their applications.
+
+**Recipe administrators** – A recipe administrator needs to manage the set of recipes registered to an environment. They must author, test, and maintain infrastructure as code modules such as Terraform or Bicep to implement the recipe. In many organizations, a recipe administrator and an environment administrator will be the same person. However, in other organizations, administration of recipes may be delegated to specialty teams for certain resource types; e.g., DBAs may maintain recipes for database resource types.
+
+## Scenarios
+
+The following scenarios will describe a final setup similar to the image below.
+
+![](2024-11-authz-feature-spec/authz.png)
+
+### Scenario 1 – Adding Radius administrators
+
+As a Radius administrator, I just installed Radius on a Kubernetes cluster and now I need to add an additional user as a Radius administrator.
+
+**User Experience**
+
+```bash
+# Radius administrative access is granted via membership in the radius-admin role in the radius-system namespace
+# Add user-1 to the radius-admin role
+kubectl create rolebinding user-1-admin --role radius-admin --user user-1 -n radius-system
+```
+
+**Result**
+
+A new RoleBinding is created in the `radius-system` namespace for `user-1` to be granted the `radius-admin` role.
+
+### Scenario 2 – Create resource group for applications
+
+As a Radius administrator, I need to create a resource group for applications and application resources to be created in as well as set permissions which define who can create applications in that group.
+
+**User Experience**
+
+```bash
+# Create application resource group 
+rad group create rg-app-1
+
+# Create developer role
+rad role definition create -f developer-role-definition.yaml
```

Comment:
Shouls this just be `rad role create` and you provide the definition either via file or flags

Reply:
It's modeled after [az role definition create](https://learn.microsoft.com/en-us/cli/azure/role/definition) and [az role assignment create](https://learn.microsoft.com/en-us/cli/azure/role/assignment)